src/client/opamArg.ml

```diff
@@ -570,8 +571,31 @@ let apply_global_options cli o =
     OpamJson.append "opam-version" (`String OpamVersion.(to_string (full ())));
     OpamJson.append "command-line"
       (`A (List.map (fun s -> `String s) (Array.to_list Sys.argv)))
-  )
-
+  );
+  (try
+     let opamfile =
+       OpamParser.FullPos.file
+         (OpamFilename.to_string
+            (OpamFile.filename
+               (OpamPath.config OpamStateConfig.(!r.root_dir))))
+     in
+     List.iter OpamParserTypes.FullPos.(function
+         | { pelem = Variable ({ pelem = "sys-pkg-manager-cmd"; _},
+                               {pelem = List { pelem = elements; _}; _}); _} ->
+           let rec aux last elements =
+             match last, elements with
+             | _, [] -> ()
+             | Some { pelem = String "cygwin"; _},
+               { pelem = String cygcheck; _}::_  ->
+               let cygbin = Filename.dirname cygcheck in
+               OpamCoreConfig.update ~cygbin ()
+             | _, element::elements -> aux (Some element) elements
+           in
+           aux None elements
+         | _ -> ())
+       opamfile.file_contents
+   with
+   | Sys_error _ | Not_found -> ())
```

Comment:
The low level version is to avoid having multiple error messages when reading with best effort.

```suggestion
  let cygbin =
    OpamStateConfig.safe_load ~lock_kind:`Lock_none
      OpamStateConfig.(!r.root_dir)
    |> OpamFile.Config.sys_pkg_manager_cmd
    |> OpamStd.String.Map.find_opt "cygwin"
    >>| OpamFilename.dirname
    >>| OpamFilename.Dir.to_string
  in
  OpamCoreConfig.update ?cygbin ();
```